pkg/manager/member/tidb_member_manager.go

```diff
@@ -712,10 +713,30 @@ func getNewTiDBSetForTidbCluster(tc *v1alpha1.TidbCluster, cm *corev1.ConfigMap)
 		slowLogFileEnvVal = slowQueryLogFile
 	}
 	envs := []corev1.EnvVar{
+		{
+			Name: "NAMESPACE",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					FieldPath: "metadata.namespace",
+				},
+			},
+		},
+		{
+			Name: "POD_NAME",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					FieldPath: "metadata.name",
+				},
+			},
+		},
 		{
 			Name:  "CLUSTER_NAME",
 			Value: tc.GetName(),
 		},
+		{
+			Name:  "HEADLESS_SERVICE_NAME",
+			Value: headlessSvcName,
+		},
```

Comment:
any changes on the statefulset will trigger rolling upgrading when people upgrade tidb-operator 1.1 from 1.0.  if we require no upgrades in upgrading operator, we should find a new way.

Reply:
like hostNetwork feature, add a switch?

Reply:
what do you think? cc @weekface @Yisaer @aylei

Reply:
OK.  I've added side effects in PR description. I'd like to add a switch if we need backward compatibility.

Reply:
another solution is that operator 1.1 doesn't apply these new changes on tidb clusters managed by operator 1.0. to achieve this, we must maintain a version label on tidb cluster CR.

Reply:
I think we could point it out it in our change log and release document.

Reply:
Actually, this behavior depends on the tidb version, which is available in the tc, can we check the version of tidb and decide if the sts.spec should be updated? @cofyc @LinuxGit

Reply:
TiDB has this flag since 2.1, I'm not sure if we have clients using the TiDB before 2.1.

Reply:
@cofyc  I think I could use https://godoc.org/github.com/coreos/go-semver/semver pkg to compare version. `latest` is not a valid semantic versioning, I need to deal with it. I'm not sure if there're other tags that can't compare. 
I could set a env variable. So startup script in Helm doesn't need to compare version.
@tennix @weekface Do you think it need to be compatible with versions before v2.1.0?

Reply:
The tidb 2.1 is not actively maintained, so I think it's ok to remove support for tidb 2.1.

Reply:
OK, the TiDB 2.1 has this flag. TiDB before 2.1 is not supported.